src/runtime/generators.js

```diff
@@ -52,6 +52,31 @@
                       state);
   }
 
+  // The following unique object serves as a non-catchable exception raised by
+  // the implementation of Generator.prototype.return
+  var SENTINEL = new Object();
+  
+  function GeneratorWrapper(iterator) {
```

Comment:
Made wrapYieldStar a function that simply returns an object literal and which does not need an extra class.